### PR TITLE
DOC: Prepare for 1.17.0rc1 release

### PR DESCRIPTION
**Pull request by charris**
Preliminary work preparing for the 1.17 release. Review of the release notes would be much appreciated. The changelog will be updated before release.

### CODE REVIEW

File: doc/release/1.17.0-notes.rst [hunk 1]
```diff
@@ -325,41 +359,39 @@ Error messages from array comparison tests such as
...
+large prime factors, pocketfft uses Bluestein's algorithm, which maintains
+``O(N log N)`` run time complexity instead of deteriorating towards ``O(N*N)``
+for prime lengths. Also, accuracy for real valued FFTs with near prime lengths
+has improved and is on par with complex valued FFTs.
 
 Performance improvements for integer sorts
 ------------------------------------------
-
 ``sort``, ``argsort``, ``ndarray.sort`` and ``ndarray.argsort`` now use radix
 sort as the default stable sort for integers and booleans. This is faster than
```
Comment (shoyer) [inline doc/release/1.17.0-notes.rst]: I thought this was only true for small integer/boolean dtypes? More generally, I  think the notes on timsort/radix sort should  probably be merged.

File: doc/release/1.17.0-notes.rst [hunk 2]
```diff
@@ -191,6 +214,17 @@ This change is backwards compatible, but now allows code like::
...
 ============
 
+New extensible random module with selectable random number generators
+---------------------------------------------------------------------
+  A new extensible random module along with four selectable random number
+  generators and improved seeding designed for use in parallel processes has
+  been added. The currently available bit generators are MT19937, PCG64,
+  Philox, and SFC64. PCG64 is the new default while MT19937 is retained for
+  backwards compatibility. Note that the legacy random module is unchanged and
+  is now frozen, your current results will not change. Extensive documentation
```
Comment (shoyer) [inline doc/release/1.17.0-notes.rst]: It would be great to dd a link here referencing the new module in our sphinx docs.

Comment (charris) [reply doc/release/1.17.0-notes.rst]: I'd like to do that, are the docs up?

Comment (mattip) [reply doc/release/1.17.0-notes.rst]: It's on devdocs, not scipy.org, but there is www.numpy.org/devdocs/reference/random/new-or-different.html

Comment (eric-wieser) [reply doc/release/1.17.0-notes.rst]: Can we just use internal sphinx links here?

File: doc/release/1.17.0-notes.rst [hunk 3]
```diff
@@ -191,6 +214,17 @@ This change is backwards compatible, but now allows code like::
 New Features
 ============
 
+New extensible random module with selectable random number generators
+---------------------------------------------------------------------
+  A new extensible random module along with four selectable random number
```
Comment (eric-wieser) [inline doc/release/1.17.0-notes.rst]: Not sure the 2 space indent will be ignored here - can you remove it?

File: doc/release/1.17.0-notes.rst [hunk 4]
```diff
@@ -325,41 +359,39 @@ Error messages from array comparison tests such as
...
-
+the old default mergesort in the majority of cases.
 
-Further improvements to ``ctypes`` support in ``np.ctypeslib``
---------------------------------------------------------------
+Further improvements to ``ctypes`` support in `numpy.ctypeslib`
+---------------------------------------------------------------
 A new `numpy.ctypeslib.as_ctypes_type` function has been added, which can be
-used to converts a `dtype` into a best-guess `ctypes` type. Thanks to this
+used to converts a ``dtype`` into a best-guess ``ctypes`` type. Thanks to this
```
Comment (eric-wieser) [inline doc/release/1.17.0-notes.rst]: As written before, doesn't `ctypes` become a sphinx link to the python docs? @mattip?

Ok with not linking `dtype`, that's pretty obvious to anyone who knows enough about numpy to read the release notes.

File: doc/release/1.17.0-notes.rst [hunk 5]
```diff
@@ -424,8 +456,8 @@ These ufuncs now call the ``__floor__``, ``__ceil__``, and ``__trunc__``
 methods when called on object arrays, making them compatible with
 `decimal.Decimal` and `fractions.Fraction` objects.
 
-`quantile` now works on `fraction.Fraction` and `decimal.Decimal` objects
--------------------------------------------------------------------------
+``quantile`` now works on ``fraction.Fraction`` and ``decimal.Decimal`` objects
```
Comment (eric-wieser) [inline doc/release/1.17.0-notes.rst]: As before, did these not create links?

File: doc/release/1.17.0-notes.rst (hunk 4 above)
Comment (charris) [reply doc/release/1.17.0-notes.rst]: I'm also curious about the automatic linking, when it work and when it doesn't. IIRC NumPy links require the full `numpy.`, but also seem to recall that that was fixed. Not sure how the links work for Python documentation. @mattip Thoughts?

Comment (rgommers) [reply doc/release/1.17.0-notes.rst]: There's normally a `currentmodule` set, and everything in that module and higher-level modules can be found. So say you're in `linalg`, then `svd` gets resolved, `linspace` as well, but `randint` needs to be referred to as `random.randint`. For release notes no currentmodule may be set, so then you'll need the full `numpy.xxx`. For Python the builtins should be resolved, anything in a stdlib module needs full import path

Comment (charris) [reply doc/release/1.17.0-notes.rst]: What does a full python import path look like?

**Pull request merged**